internal/resource/url.go

```diff
@@ -330,7 +331,14 @@ func (f *Fetcher) FetchFromS3(u url.URL, dest *os.File, opts FetchOptions) error
 	sess := f.AWSSession.Copy()
 
 	// Determine the region this bucket is in
-	region, err := s3manager.GetBucketRegion(ctx, sess, u.Host, "us-east-1")
+	regionHint, err := ec2metadata.New(sess).Region()
```

Comment:
Can this fail if this is run outside of EC2? Ignition should support downloading objects from S3 regardless of what platform it is run on.

Reply:
this would keep the default value for that case